controller/aks-cluster-config-handler.go

```diff
@@ -862,6 +866,11 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, config *aksv1.
 			logrus.Infof("Resource group [%s] updated successfully", config.Spec.ResourceGroup)
 		}
 
+		// If status is not updating, then enqueue the update (to re-enter the onChange handler)
+		if config.Status.Phase != aksConfigUpdatingPhase {
+			return h.enqueueUpdate(config)
+		}
+
```

Comment:
Should this block be inside the `if !resourceGroupExists` block a few lines up? Apologies if my previous review comment was confusing. In my opinion, we should enqueue an update right before we begin making any changes, does that make sense?